src/scheduler/fifo.c

```diff
@@ -1076,6 +1076,16 @@ end_cycle_tasks(server_info *sinfo)
 		cmp_aoename = NULL;
 	}
 
+	/* Free up conf memory which was allocated in init_config() */
+	if (conf.prime_sort != NULL)
```

Comment:
@agrawalravi90 end_cycle_tasks is called after every scheduling cycle. But these sort infos are only created when the first sched cycle runs or when scheduler is HUPed.
Deleting these here will make them dangling for the next cycle.
Please check it again